### PR TITLE
Bugfix 12063 - 500 Error when changing tag in a Contact and clicking save&close [5.x]

### LINKED ISSUE #12063: 500 Error when changing tag and save/close
### Mautic Version

4.4.x series

### PHP version

7.4.33

### What browsers are you seeing the problem on?

Chrome

### What happened?

I’m receiving a 500 error saving a contact via the GUI…

I’ve created a quick screen share to demonstrate the issue I’m facing:

[https://streamable.com/b3jzax](https://streamable.com/b3jzax)

### How can we reproduce this issue?

Step 1:  Add contact
Step 2:  Add tag to contact
Step 3: Save
Step 4: Swap tag fpr a different tag
Step 5: Save & Close

This results in a 500 response from the server.  After this the contact cannot be viewed.  The view contact page then returns a 500 error.

### Relevant log output

```shell
There is nothing in the log related to the issue described.
```

### Code of Conduct

- [x] I confirm that I have read and agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by LordRembo**
| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [ yes]
| New feature/enhancement? (use the a.x branch)      | [ ]
| Deprecations?                          | [ ]
| BC breaks? (use the c.x branch)        | [ ]
| Automated tests included?              | [ ] 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #12063 

#### Description:

This fix if for Mautic 5.x ([4.x version is here](https://github.com/mautic/mautic/pull/12105))

#### Steps to test this PR:

1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. You need a Contact that has 1 tag
3. Edit the Contact and change the tag
4. Hit Save&Close
5. You should now be able to reopen the Contact (that wasn't possible before, gave a JS error or a server error)
6. If you open the Audit log, you should see the 'old' and 'new' tag listed in the latest entry

### CODE REVIEW

**Comment by MadlenF** [comment]
Works as described!

**Review by RCheesley — DISMISSED** [review]
Thanks for the PR - works as expected and fixes the bug!

**Review by escopecz — CHANGES_REQUESTED** [review]
It looks like the 2 blocks of code are exact copies of each other. Can you please make the code DRY?

When comparing this version for M5 with the [M4 version](https://github.com/mautic/mautic/pull/12105/files) then it's quite more complicated here. Why is that?

**Comment by LordRembo** [comment]
@escopecz For 4.x I simply fixed the problem and made sure everything gets printed and formatting doesn't break. When 5.x is released we're probably not going to build further on this in 4.x and there is no (design) guide for that output. The logic is simple and effective for the requirement. 
For 5.x I split the output a bit more so different types of output can more easily be targetted/recongnized if someone decides this needs a bit more 'flair' or if at some point the data structure changes. It was easy and just a little bit of extra logic so it's nicer to work with. I didn't bother doing that for the PHP version because it doesn't need it. There probably won't be significant changes made in 4.x that would impact how nice it looks in the front-end.

**Comment by LordRembo** [comment]
Also, part of the complexity is that some logic is just shorter to do in PHP than in Twig.

**Comment by escopecz** [comment]
@LordRembo thanks for the explanation. Can you please move the 2 duplicated code sections to a separate Twig template so the code is [DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself)?

**Comment by LordRembo** [comment]
@escopecz or @mollux  I'm getting some vendor errors trying to run Mautic locally after rebasing my branch on 5.x, any ideas what could be causing this? I didn't make changes to the vendors.
```
Symfony\Component\DependencyInjection\Exception\ServiceNotFoundException: The service "fos_oauth_server.controller.authorize" has a dependency on a non-existent service "templating". - in file /vendor/symfony/dependency-injection/Compiler/CheckExceptionOnInvalidReferenceBehaviorPass.php - at line 86
/vendor/symfony/dependency-injection/Compiler/CheckExceptionOnInvalidReferenceBehaviorPass.php:86
```

**Comment by LordRembo** [comment]
In any case, I put the changes into a separate Twig template as requested

**Comment by escopecz** [comment]
> I'm getting some vendor errors trying to run Mautic locally after rebasing my branch on 5.x, any ideas what could be causing this? I didn't make changes to the vendors.

Have you tried running `composer install`?

**Comment by LordRembo** [comment]
Something messed up my database as well as the vendors. Finally fixed it by resetting, reinstalling, etc.
@escopecz It should be good for review now.

File: app/bundles/LeadBundle/Resources/views/Auditlog/_table.html.twig [hunk 1]
```diff
@@ -0,0 +1,67 @@
+<table class="table">
+    <tr>
+        <th>Field/Object</th>
+        <th>New Value</th>
+        <th>Old Value</th>
```
Comment (escopecz) [inline app/bundles/LeadBundle/Resources/views/Auditlog/_table.html.twig]: It would be nice to have these translatable.

**Review by escopecz — CHANGES_REQUESTED** [review]
It looks to me that the old and new value columns are switched
Screenshot 2023-04-14 at 10 21 45

File: app/bundles/LeadBundle/Resources/views/Auditlog/_table.html.twig (hunk 1 above)
Comment (LordRembo) [reply app/bundles/LeadBundle/Resources/views/Auditlog/_table.html.twig]: added the translations

**Comment by LordRembo** [comment]
Fixed the reversal + added translations

**Review by escopecz — APPROVED** [review]
Thank you! Looks good:
Screenshot 2023-04-17 at 10 23 16

**Pull request merged**